### DISCUSSION TITLE
Testing behavior around the passing of time

### DISCUSSION THREAD

**Discussion by tsargent** [wix/Detox#4039] [Q&A]
I am working on an app whose functionality is largely determined by the passing of time. For example, a user is asked to complete a lesson on one day. The next day, they revisit the app, and the app sees that a day has passed (determined by the device's system) and asks the user if they completed the lesson (which is some kind of activity they do away from the device). 

Is there a way to test this kind of behavior with Detox? As far as I can tell, Detox cannot change the system date. I imagine this is a common use case, so I'm wondering if there are any suggestions around how best to do this with Detox.

**Comment by jzaefferer** [comment]
Maybe use [mocking](https://wix.github.io/Detox/docs/guide/mocking/) for the module that determines the passing of time?

If the "passing of time" is determined in a backend, maybe you can instrument that to give different input to the client (the app you're testing)?
